lib/beaker/dsl/install_utils/puppet_utils.rb

```diff
@@ -583,25 +588,22 @@ def install_puppetlabs_dev_repo ( host, package_name, build_version,
                           repo_filename,
                           platform_configs_dir)
 
-            link = "%s/%s/%s/repos/%s/%s%s/PC1/%s/" %
-              [ dev_builds_url, package_name, build_version, variant,
-                fedora_prefix, version, arch ]
-
-            if not link_exists?( link )
-              link = "%s/%s/%s/repos/%s/%s%s/products/%s/" %
-                [ dev_builds_url, package_name, build_version, variant,
-                  fedora_prefix, version, arch ]
-            end
-
-            if not link_exists?( link )
-              link = "%s/%s/%s/repos/%s/%s%s/devel/%s/" %
+            link = nil
+            package_repos = options[:dev_builds_repos].nil? ? [] : [options[:dev_builds_repos]]
+            package_repos.push(['PC1', 'products', 'devel']).flatten!
```

Comment:
I agree. Ideally projects will know what collection they expect to be testing against.